python_bindings/CMakeLists.txt

```diff
@@ -0,0 +1,247 @@
+find_package(pybind11 REQUIRED)
```

Comment:
Our existing code assumes/requires that PyBind11 is installed via `pip install requirements.txt`, so this fails for me on my OSX machine. What installation process is required to make this work? Documentation and/or useful error reporting is necessary. 

Since this is an all-header library, could we perhaps use FetchContent to automate this? https://cmake.org/cmake/help/v3.14/module/FetchContent.html

Reply:
I am using a miniconda environment and the pybind11 conda package comes with the make scripts whereas the pip package doesn't.

I will look into `FetchContent` instead then.

Reply:
I'm already using FetchContent on cmake-modern